OpenRA.Mods.Cnc/FileFormats/VqaVideo.cs

```diff
@@ -542,74 +542,9 @@ void WriteBlock(int blockNumber, int count, ref int x, ref int y)
 			}
 		}
 
-		// TODO: Maybe replace this with LCWCompression.DecodeInto again later
 		public static int LCWDecodeInto(byte[] src, byte[] dest, int srcOffset = 0, bool reverse = false)
 		{
-			var ctx = new FastByteReader(src, srcOffset);
-			var destIndex = 0;
-			while (true)
-			{
-				var i = ctx.ReadByte();
-				if ((i & 0x80) == 0)
-				{
-					// case 2
-					var secondByte = ctx.ReadByte();
-					var count = ((i & 0x70) >> 4) + 3;
-					var rpos = ((i & 0xf) << 8) + secondByte;
-
-					if (destIndex + count > dest.Length)
-						return destIndex;
-
-					// Replicate previous
-					var srcIndex = destIndex - rpos;
-					if (srcIndex > destIndex)
-						throw new NotImplementedException($"srcIndex > destIndex {srcIndex} {destIndex}");
-
-					for (var j = 0; j < count; j++)
-					{
-						if (destIndex - srcIndex == 1)
-							dest[destIndex + j] = dest[destIndex - 1];
-						else
-							dest[destIndex + j] = dest[srcIndex + j];
-					}
-
-					destIndex += count;
-				}
-				else if ((i & 0x40) == 0)
-				{
-					// case 1
-					var count = i & 0x3F;
-					if (count == 0)
-						return destIndex;
-
-					ctx.CopyTo(dest, destIndex, count);
-					destIndex += count;
-				}
-				else
-				{
-					var count3 = i & 0x3F;
-					if (count3 == 0x3E)
-					{
-						// case 4
-						var count = ctx.ReadWord();
-						var color = ctx.ReadByte();
-
-						for (var end = destIndex + count; destIndex < end; destIndex++)
-							dest[destIndex] = color;
-					}
-					else
-					{
-						// If count3 == 0x3F it's case 5, else case 3
-						var count = count3 == 0x3F ? ctx.ReadWord() : count3 + 3;
-						var srcIndex = reverse ? destIndex - ctx.ReadWord() : ctx.ReadWord();
-						if (srcIndex >= destIndex)
-							throw new NotImplementedException($"srcIndex >= destIndex {srcIndex} {destIndex}");
-
-						for (var end = destIndex + count; destIndex < end; destIndex++)
-							dest[destIndex] = dest[srcIndex++];
-					}
-				}
-			}
+			return LCWCompression.DecodeInto(src, dest, srcOffset, reverse);
```

Comment:
This wrapper shouldn't exist. We should revert to something like https://github.com/OpenRA/OpenRA/commit/c687600d6610bd2c163eb0223b5f06ec0857eb8a

Reply:
Ok, reverted the changes in this PR.

Reply:
This is not what I meant. I meant that `LCWCompression.DecodeInto` is equivalent to `LCWDecodeInto` and thus the code should instead use `LCWCompression.DecodeInto`

Reply:
Ah ok, updated.